File: src/ELibrary.Backend/UserApi/Domain/Dtos/Responses/AdminUserResponse.cs
using UserEntities.Domain.Entities;

namespace UserApi.Domain.Dtos.Responses
{
    public class AdminUserResponse
    {
        public string Id { get; set; } = string.Empty;
        public string UserName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public DateTime RegistredAtUtc { get; set; }
        public DateTime UpdatedAtUtc { get; set; }
        public List<string> Roles { get; set; } = [];
        public List<AuthenticationMethod> AuthenticationMethods { get; set; } = [];

    }
}

File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/user/responses/adminUserResponse.ts
import { AdminUser, AuthenticationMethod } from "../../../..";

export interface AdminUserResponse {
    id: string;
    userName: string;
    email: string;
    registredAtUtc: Date;
    updatedAtUtc: Date;
    roles: string[];
    authenticationMethods: AuthenticationMethod[];
}

export function mapAdminUserResponseToAdminUser(response: AdminUserResponse): AdminUser {
    return {
        id: response.id,
        userName: response.userName,
        email: response.email,
        registredAt: new Date(response.registredAtUtc),
        updatedAt: new Date(response.updatedAtUtc),
        roles: response.roles,
        authenticationMethods: response.authenticationMethods
    }
}